docs/tutorials/shop-page.md

```diff
@@ -0,0 +1,396 @@
+
+```lua
+function ShopPage:render()
+	local props = self.props
+
+	local items = props.items
+	local frameProps = props.frameProps
+	local padding = props.padding
+
+	local cellSize = state.cellSize
+	local canvasHeight = state.canvasHeight
+
+	-- we need to clone the frameProps because we can't mutate it!
+	local newFrameProps = {}
+
+	for key, value in pairs(frameProps) do
+		newFrameProps[key] = value
+	end
+
+	-- now that we have cloned the scrolling frame props, we add new entries
+	newFrameProps.CanvasSize = UDim2.new(0, 0, 0, canvasHeight)
+
+	return Roact.createElement("ScrollingFrame", newFrameProps, {
+		UIGrid = Roact.createElement("UIGridLayout", {
+			CellPadding = UDim2.new(0, padding, 0, padding),
+			-- now we can use the cellSize from the state!
+			CellSize = UDim2.new(0, cellSize, 0, cellSize * itemAspectRatio),
+			HorizontalAlignment = Enum.HorizontalAlignment.Center,
+			VerticalAlignment = Enum.VerticalAlignment.Center,
+			SortOrder = Enum.SortOrder.LayoutOrder,
+		}),
+	})
+end
+```
+
+Then, we need to update the state when the scrolling frame size changes. To connect a function to a changed property event, we use [`Roact.Change.NameOfProperty`](https://roblox.github.io/roact/api-reference/#roactchange). In our case, we will listen to the `AbsoluteSize` property change of the scrolling frame. The cell size of the grid will be computed to maximize the horizontal space. After, the aspect ratio of the items is used to find the vertical space needed to contain all the items.
+
+```lua
+function ShopPage:init()
+	...
+
+	self.onAbsoluteSizeChanged = function(frame)
+		local props = self.props
+		local padding = props.padding
+		local itemsPerRow = props.itemsPerRow
+
+		local totalWidth = frame.AbsoluteSize.X
+		local cellWidth = (totalWidth - padding * (itemsPerRow + 1)) / itemsPerRow
+		local cellHeight = cellWidth * props.itemAspectRatio
+		local rows = math.ceil(#props.items / itemsPerRow)
+		local canvasHeight = rows * cellHeight + padding * (rows + 1)
+
+		-- we update the state with the new values
+		self:setState({
+			cellSize = cellWidth,
+			canvasHeight = canvasHeight,
+		})
+	end
+end
+
+function ShopPage:render()
+	...
+	-- connect the function in the render method
+	newFrameProps[Roact.Change.AbsoluteSize] = self.onAbsoluteSizeChanged
+	...
+end
+```
+
+## Adding Items
+
+In the previous section we created a scrolling frame that contains a UIGridLayout object. Now, we want to add elements that represent each item we sell in our shop. In order to be able to add a variable amount of items in our component, we will iterate on the list of products to create an element for each of them.
+
+To keep our component simple, let's divide this feature into different components:
+ - **ProductItem:** a card that shows the price of an item with its image and when clicked prompt the player to buy the product
+ - **ProductItemList:** iterate on the list of items to create each individual ProductItem element
```

Comment:
I think you could name these `Item` and `ItemList`. `ProductItem` sounds like you're saying the same thing twice and is a little clunky to read